infrastructure/src/main/java/org/corfudb/infrastructure/BatchProcessor.java

```diff
@@ -237,13 +237,11 @@ private void process() {
                     lastOp = currentOp;
                 }
             }
-        } catch (Exception e) {
-            log.error("Caught exception in the write processor ", e);
-            context.setErrorStatus();
         } catch (Throwable th) {
+            // Initiate a shutdown of the JVM when an unhandled Exception/Throwable is
+            // encountered, including IOExceptions during fsync and OutOfDirectMemoryErrors.
             log.error("Encountered throwable in the write processor ", th);
-            context.setErrorStatus();
-            throw th;
+            System.exit(CorfuServer.EXIT_ERROR_CODE);
```

Comment:
Can the feature be tested (system exit on BP Exceptions) with a UT or IT?

Reply:
Done! Validated that system exit is called with the appropriate code in a UT.